### ISSUE TITLE
Again, more of a question than an issue

### ISSUE THREAD

**Issue by sumitwy** [klundberg/Weakify#4]
Can weakify be used for blocks also?

Example:
````
self.syncEvents(completion: {
    self.syncEventsAgain()
})
````

In this case, the completion handler passed to `syncEvents` method captures self strongly. So self cannot be deallocated unless this handler is executed. The conventional method to fix this is to capture self weakly and then do something like

```
self.syncEvents(completion: { [weak self] in
  if self != nil {
      self!.syncEventsAgain()
  }
})
```

 But then it's not fun :)
Can we modify weakify or write something similar so that it can work with this case also with the same brevity?

**Comment by dasMajo** [comment]
You can just do

**Comment by dasMajo** [comment]
You can just do
```
self.syncEvents(completion: { [weak self] in
     self?.syncEventsAgain()
})
```
No need for weakify

**Comment by klundberg** [comment]
if you wanted to use weakify with your original example @sumitwy, you could do this:

```swift
self.syncEvents(completion: weakify(self, YourType.syncEventsAgain))
```

But if your closure has multiple lines then weakify won't quite work as well :) If the type of `self` is very long, then referencing the function may be too wordy and you'd be better served with `[weak self]` inside a closure.

In general, inline calls might be better to just use a closure. `weakify` may be nicer if you have to assign the function to a property, since in my opinion reading function call syntax is a little nicer than closure syntax when written on a single line, but to each their own!
